test/sqllogictest/transform/table_elimination.slt

```diff
@@ -0,0 +1,19 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
```

Comment:
The optimization does not kick in here even though it the union of an input and its negation is present:

```
materialize=> EXPLAIN SELECT * FROM t1 AS a1 LEFT JOIN t1 AS a2 ON (a1.key = a2.key) WHERE a1.nokey = 1;
          Optimized Plan           
-----------------------------------
 %0 =                             +
 | Get materialize.public.t1 (u75)+
 | Filter (#1 = 1)                +
 | Negate                         +
                                  +
 %1 =                             +
 | Get materialize.public.t1 (u75)+
 | Filter (#1 = 1)                +
                                  +
 %2 =                             +
 | Union %0 %1                    +
 | Map null, null                 +
                                  +
 %3 =                             +
 | Get materialize.public.t1 (u75)+
 | Filter (#1 = 1)                +
 | Project (#0, #1, #0, #1)       +
                                  +
 %4 =                             +
 | Union %2 %3                    +
```

Reply:
It does work now:

```
query T multiline
EXPLAIN SELECT * FROM t1 AS a1 LEFT JOIN t1 AS a2 ON (a1.key = a2.key) WHERE a1.nokey = 1
----
%0 =
| Get materialize.public.t1 (u1)
| Filter (#1 = 1)
| Project (#0, #1, #0, #1)

EOF

query IIII
SELECT * FROM t1 AS a1 LEFT JOIN t1 AS a2 ON (a1.key = a2.key) WHERE a1.nokey = 1
----
1  1  1  1
```